docs/observability/authentication.md

```diff
@@ -0,0 +1,102 @@
+automatically via
+[Application Default Credentials](https://cloud.google.com/docs/authentication/provide-credentials-adc).
+
+You will need to apply the following roles to the service account that is 
+running your code (i.e. 'attached service account') via the 
+[IAM Console](https://console.cloud.google.com/iam-admin/iam):
+
+- `roles/monitoring.metricWriter`
+- `roles/cloudtrace.agent`
+- `roles/logging.logWriter`
+
+To find your default service account:
+
+1. Navigate to the [service accounts page](https://console.cloud.google.com/iam-admin/serviceaccounts) in the Google Cloud Console
+2. Select your project
+3. Your default service account should look like *project-number*-compute@developer.gserviceaccount.com or *project-id*@appspot.gserviceaccount.com if you are using App Engine.
+
+## Deploying outside of Google Cloud with Application Default Credentials (ADC)
+
+If possible, it is still recommended to leverage the 
+[Application Default Credentials](https://cloud.google.com/docs/authentication/provide-credentials-adc) 
+process to make credentials available to the plugin.
+
+Typically this involves generating a service account key/pair and deploying
+those credentials to your production environment. 
+
+1. Follow the instructions to set up a 
+[service account key](https://cloud.google.com/iam/docs/keys-create-delete#creating).
+
+2. Ensure the service account has the following roles:
+    - `roles/monitoring.metricWriter`
+    - `roles/cloudtrace.agent`
+    - `roles/logging.logWriter`
+
+3. Deploy the credential file to production (**do not** check into source code)
+
+4. Set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable as the path to
+the credential file.
+
+    ```
+    GOOGLE_APPLICATION_CREDENTIALS = "path/to/your/key/file"
+    ```
+
+## Deploying outside of Google Cloud without Application Default Credentials (ADC)
+
+In some serverless environments, you may not be able to deploy a credential
+file.
+
+1. Follow the instructions to set up a 
+[service account key](https://cloud.google.com/iam/docs/keys-create-delete#creating).
+
+2. Ensure the service account has the following roles:
+    - `roles/monitoring.metricWriter`
+    - `roles/cloudtrace.agent`
+    - `roles/logging.logWriter`
+
+3. Download the credential file.
+
+4. Assign the contents of the credential file to the `GCLOUD_SERVICE_ACCOUNT_CREDS` environment variable as follows:
+
+```
+GCLOUD_SERVICE_ACCOUNT_CREDS='{
+  "type": "service_account",
+  "project_id": "your-project-id",
+  "private_key_id": "your-private-key-id",
+  "private_key": "your-private-key",
+  "client_email": "your-client-email",
+  "client_id": "your-client-id",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://accounts.google.com/o/oauth2/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "your-cert-url"
+}'
```

Comment:
From my experience, projectId will also need to be passed into the configuration options, or via env variable. PR incoming to plumb that through properly.